user/tests/wiring/no_fixture/pwm.cpp

```diff
@@ -82,14 +82,33 @@ test(PWM_AnalogWriteOnPinResultsInCorrectAnalogValue) {
     //To Do : Add test for remaining pins if required
 }
 
-test(PWM_FastAnalogWriteOnPinResultsInCorrectAnalogValue) {
+test(PWM_AnalogWriteWithFrequencyOnPinResultsInCorrectFrequency) {
+    // when
+    pinMode(pin, OUTPUT);
+    analogWrite(pin, 150, 10000);
+    // then
+    assertEqual(HAL_PWM_Get_Frequency(pin), 10000);
+    //To Do : Add test for remaining pins if required
+}
+
+test(PWM_AnalogWriteWithFrequencyOnPinResultsInCorrectAnalogValue) {
+    // when
+    pinMode(pin, OUTPUT);
+    analogWrite(pin, 200, 10000);
+    // then
+    assertEqual(HAL_PWM_Get_AnalogValue(pin), 200);
+    //To Do : Add test for remaining pins if required
+}
+
+
+test(PWM_LowDCAnalogWriteOnPinResultsInCorrectPulseWidth) {
     pin_t pin = D0; // pin under test
     // when
     pinMode(pin, OUTPUT);
 
+    analogWrite(pin, 25); // 10% Duty Cycle at 500Hz = 200us HIGH, 1800us LOW.
```

Comment:
I believe this was originally in the for() loop to help vet issues with re-initializing the PWM.  Having it in the for() loop should not cause a problem, and at least one of these tests should keep it in the for() loop :)